File: backend/app/schemas/vote.py
from pydantic import BaseModel
from datetime import datetime

class VoteBase(BaseModel):
    feature_id: int

class VoteCreate(VoteBase):
    pass

class Vote(VoteBase):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: mobile/FeatureVoting/Models/Vote.swift
import Foundation

struct Vote: Codable, Identifiable {
    let id: Int
    let userId: Int
    let featureId: Int
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case featureId = "feature_id"
        case createdAt = "created_at"
    }
}

struct VoteCreate: Codable {
    let featureId: Int

    enum CodingKeys: String, CodingKey {
        case featureId = "feature_id"
    }
}